tests/test_visualization/test_boxplot.py

```diff
@@ -0,0 +1,66 @@
+import unittest
+import pandas as pd
+import anndata
+import matplotlib
+import matplotlib.pyplot as plt
+from spac.visualization import boxplot
+matplotlib.use('Agg')  # Set the backend to 'Agg' to suppress plot window
+
+
+class TestBoxplot(unittest.TestCase):
+
+    def setUp(self):
+        """Set up testing environment."""
+        X = pd.DataFrame({
+            'feature1': [1, 3, 5, 7],
+            'feature2': [2, 4, 6, 8],
+        })
+
+        annotation = pd.DataFrame({
+            'phenotype': [
+                'phenotype1',
+                'phenotype1',
+                'phenotype2',
+                'phenotype2'
+            ]
+        })
+
+        self.adata = anndata.AnnData(X=X, obs=annotation)
+
+    def test_returns_correct_types(self):
```

Comment:
@fangliu117 , please add a test to make sure that the annotation mode is working as expected.